### ISSUE TITLE
Media Capabilities data dictionary requires properties not defined in the manifest

### ISSUE THREAD

**Issue by wilaw** [Dash-Industry-Forum/DASH-IF-IOP#385]
The Media Capabilities API https://w3c.github.io/media-capabilities/ for MSE-based players defines a VideoConfiguration https://w3c.github.io/media-capabilities/#dictdef-videoconfiguration with the following properties:

dictionary VideoConfiguration {
  required DOMString contentType;
  required unsigned long width;
  required unsigned long height;
  required unsigned long long bitrate;
  required double framerate;
  boolean hasAlphaChannel;
  **HdrMetadataType hdrMetadataType;
  ColorGamut colorGamut;
  TransferFunction transferFunction;**
};

The last three are defined by enumerated properties which are not currently conveyed in the DASh manifest. 

enum HdrMetadataType {
  "smpteSt2086",
  "smpteSt2094-10",
  "smpteSt2094-40"
};

enum ColorGamut {
  "srgb",
  "p3",
  "rec2020"
};

enum TransferFunction {
  "srgb",
  "pq",
  "hlg"
};

Request is for DASH IF to define a standard means of defining these properties within a DASH manifest,

**Comment by haudiobe** [comment]
This has already been completed in v4.3.

 

There are supplemental descriptors for this.

 

Thomas

**Comment by haudiobe** [comment]
More offline discussion happening. Let's use Codec Independent Code Points.

**Comment by jpiesing** [comment]
DVB-DASH uses CICP.

**Comment by haudiobe** [comment]
Yes CICP is good for most of it. HDR metadata is not covered in CICP though.

 

/T

**Comment by ZmGorynych** [comment]
SMPTE 2086 is a "base layer", both DolbyVision and HDR10+ require it for PQ, afaik.
It may be worth splitting metadata into dynamic (DolbyVision, HDR10+) and static. You will have streams that will have smpteSt2086, smpteSt2094-10, and smpteSt2094-40 metadata at the same time. 2094-unaware decoders will just process the stream as HDR10.
I'd add a new supplemental property with a list of HDR metadata types carried in the stream.
`<SupplementalProperty schemeIdUri="xxx:hdr-metadata:2020` value="smpteSt2086  smpteSt2094-10"/>`

Another potential issues is that there are multiple versions of DolbyVision (unsure about HDR10+), and you need to know which version is supported. This currently shows in the codecs string
